kubeconfig.go

```diff
@@ -83,38 +81,47 @@ func (k KubeConfig) NewG8sClientForApp(ctx context.Context, app v1alpha1.App) (v
 // the current cluster is returned.
 func (k KubeConfig) NewK8sClientForApp(ctx context.Context, app v1alpha1.App) (kubernetes.Interface, error) {
 	secretName := secretName(app)
+	secretNamespace := secretNamespace(app)
 
 	// KubeConfig is not configured so connect to current cluster.
 	if secretName == "" {
 		return k.k8sClient, nil
 	}
 
-	kubeConfig, err := k.getKubeConfigFromSecret(ctx, secretName, secretNamespace(app))
+	restConfig, err := k.NewRESTConfigForApp(ctx, secretName, secretNamespace)
 	if err != nil {
-		return nil, err
+		return nil, microerror.Mask(err)
 	}
 
-	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfig)
+	client, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
 		return nil, err
 	}
+	return client, nil
+}
+
+func (k KubeConfig) NewRESTConfigForApp(ctx context.Context, secretName, secretNamespace string) (*restclient.Config, error) {
```

Comment:
In general I'd question the need of other methods when we have `*rest.Config` available. This is just a matter of calling `NewForConfig` as needed.

Reply:
Since `app-operator` already used other two methods, we planned to phase out other two after merging this. WDYT?

Reply:
👍